### PR TITLE
Improve: disable conf in files and attributes

### PR DESCRIPTION
**Pull request by gpetiot**
From https://github.com/janestreet/ocamlformat/commit/1af58007af7f8627133b14fa47414b9180a2dcc9

### CODE REVIEW

File: src/Conf.ml [hunk 1]
```diff
@@ -1509,13 +1519,21 @@ let parse_line config ~from s =
               , Format.sprintf "expecting %s but got %s" Version.version
                   value ))
     | name, `File x ->
-        C.update ~config
-          ~from:(`Parsed (`File x))
-          ~name ~value ~inline:false
+        if !disable_conf_files then
+          Error (`Malformed "Disabled file configuration of ocamlformat")
```
Comment (Julow) [inline src/Conf.ml]: This will trigger an error when there is a `.ocamlformat` file in the current directory.

Comment (Julow) [reply src/Conf.ml]: Also the error message is pretty ugly:

```
Fatal error: exception ("User Error: malformed .ocamlformat file"
  ("invalid format" "Disabled file configuration of ocamlformat")
  ("invalid format" "Disabled file configuration of ocamlformat")
  ("invalid format" "Disabled file configuration of ocamlformat"))
```
(followed by stack trace)

File: src/Conf.ml [hunk 2]
```diff
@@ -1509,13 +1519,21 @@ let parse_line config ~from s =
...
+        else
+          C.update ~config
+            ~from:(`Parsed (`File x))
+            ~name ~value ~inline:false
     | name, `Attribute ->
-        C.update ~config
-          ~from:(`Parsed `Attribute)
-          ~name ~value ~inline:true
+        if !disable_conf_attrs && not (String.equal name "disable") then
+          Error
```
Comment (Julow) [inline src/Conf.ml]: This error is not shown to the user and attributes are ignored.

**Comment by gpetiot** [comment]
> What is the motivation for these options ?

The motivation is to not have configuration files and attributes silently overriding the preset configuration (here by Janestreet but it could be useful to others)

File: src/Conf.ml (hunk 1 above)
Comment (gpetiot) [reply src/Conf.ml]: fixed

File: src/Conf.ml (hunk 2 above)
Comment (gpetiot) [reply src/Conf.ml]: fixed

File: src/Conf.ml [hunk 3]
```diff
@@ -1513,9 +1523,18 @@ let parse_line config ~from s =
           ~from:(`Parsed (`File x))
           ~name ~value ~inline:false
     | name, `Attribute ->
-        C.update ~config
-          ~from:(`Parsed `Attribute)
-          ~name ~value ~inline:true
+        if
+          !disable_conf_attrs
+          && (not (String.equal name "disable"))
+          && not config.quiet
```
Comment (Julow) [inline src/Conf.ml]: If `--quiet` is passed, attributes are not ignored. You meant a nested `if` ?

File: src/Conf.ml [hunk 4]
```diff
@@ -1519,17 +1519,18 @@ let parse_line config ~from s =
...
+          ~from:(`Parsed (`File x))
+          ~name ~value ~inline:false
     | name, `Attribute ->
-        if !disable_conf_attrs && not (String.equal name "disable") then
-          Error
-            (`Malformed
-              "Disabled configuration of ocamlformat in attributes")
+        if
+          !disable_conf_attrs
+          && (not (String.equal name "disable"))
```
Comment (hhugo) [inline src/Conf.ml]: This is very janestreet related, I don't think we should specialize "disable" just yet.

File: src/Conf.ml (hunk 3 above)
Comment (gpetiot) [reply src/Conf.ml]: good catch

**Pull request merged**